src/litdata/processing/functions.py

```diff
@@ -470,3 +482,112 @@ def __iter__(self) -> Any:
                         future = executor.submit(_listdir, folder)
                         self.futures.append(future)
         return
+
+
+@dataclass
+class CopyInfo:
+    input_dir: Dir
+    old_filename: str
+    new_filename: str
+
+
+def merge_datasets(input_dirs: List[str], output_dir: str) -> None:
+    """The merge_datasets utility enables to merge multiple existing optimized datasets into a single optimized
+    dataset.
+
+    Arguments:
+        input_dirs: A list of directories pointing to the existing optimized datasets.
+        output_dir: The directory where the merged dataset would be stored.
```

Comment:
can the output dir be one of the input dirs as well? If yes, would it just append? That'd be great!

If no, why not? Add a check here maybe?

Reply:
Not supported right now, but we can add support for it in another PR.